### PR TITLE
BUG: read_excel forward-filling MI names

### LINKED ISSUE #34673: BUG: read_excel: incorrect multi-index values - 2
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import pandas as pd
df = pd.read_excel('test.xlsx', header=[0, 1], index_col=[0, 1])
print(df.index)
```
(`engine` could be `xlrd` or `openpyxl`, doesn't matter.)

[test.xlsx](https://github.com/pandas-dev/pandas/files/4753954/test.xlsx)
image

#### Problem description

```
MultiIndex([('x', 'B')],
           names=['A', 'B'])
```

#### Expected Output

```
MultiIndex([('x', None)],
           names=['A', 'B'])
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.7.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.42-calculate
machine          : x86_64
processor        : Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz
byteorder        : little
LC_ALL           : None
LANG             : ru_RU.utf8
LOCALE           : ru_RU.UTF-8

pandas           : 1.0.3
numpy            : 1.18.1
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 46.4.0.post20200518
Cython           : 0.29.17
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : 1.2.8
lxml.etree       : 4.5.0
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.13.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.5.0
matplotlib       : 3.1.3
numexpr          : None
odfpy            : None
openpyxl         : 3.0.3
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
xlsxwriter       : 1.2.8
numba            : None
</details>

### PR DESCRIPTION
**Pull request by mzeitlin11**
- [x] closes #34673
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Added test with OP's issue as a sheet in the style of existing sheets in `tests/io/data/excel/testmultiindex.*`

(the same as the "both_name" sheet, but with the cell below the MI name empty)

New sheet looks like:
Screen Shot 2020-12-15 at 9 50 02 PM

In master without this change, the empty cell was being filled with the index name `ilvl2`

### CODE REVIEW

File: pandas/tests/io/excel/test_readers.py [hunk 1]
```diff
@@ -804,6 +804,19 @@ def test_read_excel_multiindex(self, read_ext):
         )
         tm.assert_frame_equal(actual, expected)
 
+        # GH34673: "both_name_blank_after_mi_name" sheet
```
Comment (gfyoung) [inline pandas/tests/io/excel/test_readers.py:807]: I would make this a separate test.  There isn't sufficient overlap (besides `mi_file`).

Comment (mzeitlin11) [reply pandas/tests/io/excel/test_readers.py:845]: Extracted into new test

File: pandas/io/excel/_base.py [hunk 2]
```diff
@@ -513,6 +515,11 @@ def parse(
                 else:
                     offset = 1 + max(header)
 
+                # GH34673: if MultiIndex names present and not defined in the header,
+                # don't forward fill the names
+                if has_index_names:
+                    offset += 1
```
Comment (WillAyd) [inline pandas/io/excel/_base.py:521]: I could be wrong but the +1 feels a little arbitrary here - what would happen if there were say two consecutive rows in the column with missing data?

Comment (mzeitlin11) [reply pandas/io/excel/_base.py:521]: Good point, thanks for bringing this up. Will add another test for this, but I think the +1 is ok here. It's not about handling 1 empty row, the problem was that `offset` was the line containing MI names when index names were present. `offset` is the position from which forward filling logic started, so the names were being used to forward fill missing MI values. The +1 is to make sure that `offset` points to the first line of MI values instead, so forward filling only includes actual values.

**Review by jreback — APPROVED** [review]
lgtm. @WillAyd @gfyoung good here?

File: pandas/io/excel/_base.py (hunk 2 above)
Comment (gfyoung) [reply pandas/io/excel/_base.py:521]: This explanation (paraphrased) would be good to include as a comment.

Comment (mzeitlin11) [reply pandas/io/excel/_base.py:522]: Added test for multiple blanks after name and a clearer comment

**Review by WillAyd — APPROVED** [review]
lgtm @gfyoung merge whenever you are good with it

**Review by gfyoung — APPROVED** [review]

**Comment by gfyoung** [comment]
Thanks @mzeitlin11 !

**Comment by kuraga** [comment]
@mzeitlin11 , @gfyoung , @jreback , @WillAyd ,
should we document such behavior (taking #47496 into account, too) in `read_excel`?

And, more: do `read_csv`/etc. do the same?

**Pull request merged**